SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -232,7 +228,7 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 			FClassInfo* Info = TypebindingManager->FindClassInfoByClass(Actor->GetClass());
 			check(Info);
 
-			if (Op.component_id == Info->SchemaComponents[SCHEMA_ClientRPC])
+			if ((Actor->IsA<APawn>() || Actor->IsA<APlayerController>()) && Op.component_id == Info->SchemaComponents[SCHEMA_ClientRPC])
```

Comment:
Is it not safe (and less repetitive) to say `if (Actor->RemoteRole = ROLE_AutonomousProxy)` here?

Reply:
I think we also need to handle the case where the client loses authority over the clientRPC component, however rare that might be. In this case I think we'd want to switch back to SimulatedProxy, which wouldn't happen if that check was in the if statement.